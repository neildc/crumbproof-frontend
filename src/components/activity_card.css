.cardPhoto{
    background: #f0f0f0;
    min-height:300px !important;
    width: 100%;
    heigth: 100%;
}

.menuButton {
    float:right;
}
