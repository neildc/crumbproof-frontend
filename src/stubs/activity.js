export const simpleActivity = {
  id: 11,
  name: 'Saturday wholewheat but on a Thursday (first attempt)',
  user: 'neildc',
  recipe: {
    id: 5,
    user: 'neildc',
    data: {
      name: 'The Saturday 75% Whole Wheat Bread',
      ingredients: [
        {
          quantity: 750,
          id: 'eced125b-b723-445e-bc0b-80e361afac89',
          unit: 'g',
          name: 'Whole wheat flour',
        },
        {
          quantity: 3,
          id: '822a69a9-f26c-4ecb-ab67-8957ea32d16a',
          unit: 'g',
          name: 'Instant dried yeast',
        },
      ],
      yield_count: 2,
      credits: 'Flour water salt yeast by Ken Forkish',
      yield_type: 'Loaves',
      room_temperature: 22,
      instructions: [
        {
          content: 'Mix the all the flour and water',
          id: '08bde909-e991-4297-9d3f-366c07f0db5d',
        },
        {
          content: 'Cover and rest (up to 30 minutes)',
          id: 'd056221d-2403-4e1e-af6b-2047dc20fa2b',
          time_gap_to_next: 20,
        }
      ],
    },
    diff: null,
    base_recipe: null,
    parent: null,
    created: '2017-10-31T22:18:58.628723Z',
  },
  started: null,
  created: '2017-11-09T03:23:13.595165Z',
  completed: null,
  crumb_shot: 'https://crumbproof-img.s3.amazonaws.com/media/images/8fa50806-f37.jpg',
  notes: "Boule didn't hold much tension after the final proof. Decent crumb but crust wasn't crunchy.\n\nMaybe less water next time.",
};
