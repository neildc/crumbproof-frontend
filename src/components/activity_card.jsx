import React, {Component} from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { deleteActivity } from "../actions/actions_activity";
import LinearProgress from 'material-ui/LinearProgress';
import {Card, CardMedia, CardTitle} from 'material-ui/Card';
import moment from "moment";
import ImageZoom from 'react-medium-image-zoom'
import DeleteIcon from 'material-ui/svg-icons/action/delete';
import TripleDotIcon from 'material-ui/svg-icons/navigation/more-horiz';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import IconButton from 'material-ui/IconButton';
import LazyLoad from "react-lazyload"


import RecipeDiff from "./recipe_diff";
import "./activity_card.css";


class ActivityCard extends Component {

  handleDeleteClick() {
    this.props.deleteActivity(this.props.activity.id,() => {})
  }

  render() {

    const { activity } = this.props;

    if (!activity) {
      return <LinearProgress mode="indeterminate" />;
    }

    var duration = null;

    if (activity.completed && activity.started) {
      duration = moment(activity.completed).diff(moment(activity.started), 'hours');
    }

    const created = moment(activity.created).fromNow();

    return (
      <Card >

        <CardMedia
          className="cardPhoto"
          /*
          *  Without this the overlay captures all click events
          *  instead of the events being handled by the ImageZoom component
          */
          overlayContainerStyle={{pointerEvents:"none"}}
          overlay={
            <CardTitle
              title={activity.name}
              subtitle={`${created} by ${activity.user}`}
            />
          }>

          <LazyLoad offset={200}>
            <ImageZoom
              image={{
                src: activity.crumb_shot,
                alt: '',
                className: 'img',
                style: { maxWidth: '100%' }
              }}
            />
          </LazyLoad>
        </CardMedia>

        <div style={{padding:"30px"}}>

          {(this.props.user === activity.user) &&
            <IconMenu
              iconButtonElement={<IconButton tooltip="Menu"><TripleDotIcon/></IconButton>}
              anchorOrigin={{horizontal: 'left', vertical: 'top'}}
              targetOrigin={{horizontal: 'left', vertical: 'top'}}
              className={"menuButton"}
            >
              <MenuItem
                primaryText="Delete"
                leftIcon={<DeleteIcon color="black"/>}
                onClick={this.handleDeleteClick.bind(this)}
              />
            </IconMenu>
          }

          { activity.recipe &&
              <p>
              <b>Recipe Used:</b> <Link to={`/recipes/${activity.recipe.id}`}>{activity.recipe.data.name}</Link>
              </p>
          }
          {activity.oven_start && activity.oven_end &&
            <p>In the oven: {moment(activity.oven_start).format('h:mm:ss a')} {" - "}
              {moment(activity.oven_end).format('h:mm:ss a')}
            </p>
          }

          { activity.completed &&
            <p>Completed: {activity.completed}</p>
          }

          { duration &&
            <p>Total duration: {duration} hours</p>
          }

          { activity.notes &&
            <p><b>Notes:</b> {activity.notes}</p>
          }

          { activity.recipe &&
            <RecipeDiff
              recipe={this.props.activity.recipe}
            />
          }
        </div>

      </Card>
    )
  }
}

function mapStateToProps({ auth }) {
  return {
    user: auth.user
  };
}

export default connect(mapStateToProps, { deleteActivity })(ActivityCard);
