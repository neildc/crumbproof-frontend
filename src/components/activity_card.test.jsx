import React from 'react';
import { shallow } from 'enzyme';
import { ActivityCard } from './activity_card';
import { simpleActivity } from '../stubs/activity';

it('renders without throwing', () => {
  const component = shallow(
    <ActivityCard
      activity={simpleActivity}
      user="test"
    />,
  );

  expect(component).toMatchSnapshot();
});
