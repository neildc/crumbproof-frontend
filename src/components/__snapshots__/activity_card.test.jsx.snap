// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without throwing 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
>
  <CardMedia
    className="cardPhoto"
    overlay={
      <CardTitle
        subtitle="in 5 months by neildc"
        title="Saturday wholewheat but on a Thursday (first attempt)"
      />
    }
    overlayContainerStyle={
      Object {
        "pointerEvents": "none",
      }
    }
  >
    <LazyLoad
      offset={200}
      once={true}
      overflow={false}
      resize={false}
      scroll={true}
      unmountIfInvisible={false}
    >
      <ImageZoom
        defaultStyles={
          Object {
            "image": Object {},
            "overlay": Object {},
            "zoomContainer": Object {},
            "zoomImage": Object {},
          }
        }
        image={
          Object {
            "alt": "",
            "className": "img",
            "src": "https://crumbproof-img.s3.amazonaws.com/media/images/8fa50806-f37.jpg",
            "style": Object {
              "maxWidth": "100%",
            },
          }
        }
        onUnzoom={[Function]}
        onZoom={[Function]}
        shouldHandleZoom={[Function]}
        shouldReplaceImage={true}
        shouldRespectMaxDimension={false}
        zoomMargin={40}
      />
    </LazyLoad>
  </CardMedia>
  <div
    style={
      Object {
        "padding": "30px",
      }
    }
  >
    <div>
      <p>
        <b>
          Recipe Used:
        </b>
         
        <Link
          replace={false}
          to="/recipes/5"
        >
          The Saturday 75% Whole Wheat Bread
        </Link>
      </p>
      <p>
        <b>
          Notes:
        </b>
         
        Boule didn't hold much tension after the final proof. Decent crumb but crust wasn't crunchy.
        
        Maybe less water next time.
      </p>
      <Connect(RecipeDiff)
        recipe={
          Object {
            "base_recipe": null,
            "created": "2017-10-31T22:18:58.628723Z",
            "data": Object {
              "credits": "Flour water salt yeast by Ken Forkish",
              "ingredients": Array [
                Object {
                  "id": "eced125b-b723-445e-bc0b-80e361afac89",
                  "name": "Whole wheat flour",
                  "quantity": 750,
                  "unit": "g",
                },
                Object {
                  "id": "822a69a9-f26c-4ecb-ab67-8957ea32d16a",
                  "name": "Instant dried yeast",
                  "quantity": 3,
                  "unit": "g",
                },
              ],
              "instructions": Array [
                Object {
                  "content": "Mix the all the flour and water",
                  "id": "08bde909-e991-4297-9d3f-366c07f0db5d",
                },
                Object {
                  "content": "Cover and rest (up to 30 minutes)",
                  "id": "d056221d-2403-4e1e-af6b-2047dc20fa2b",
                  "time_gap_to_next": 20,
                },
              ],
              "name": "The Saturday 75% Whole Wheat Bread",
              "room_temperature": 22,
              "yield_count": 2,
              "yield_type": "Loaves",
            },
            "diff": null,
            "id": 5,
            "parent": null,
            "user": "neildc",
          }
        }
      />
    </div>
  </div>
</Card>
`;
